cpp/pipelines/add_subsystem_tags.cc

```diff
@@ -425,27 +424,100 @@ void AddSubsystemTags(MARC::Reader * const marc_reader, MARC::Writer * const mar
 }
 
 
+void ExtractAuthors(MARC::Reader * const marc_reader, std::map<std::string, std::set<std::string>> * authors,
+                        const std::unordered_set<std::string> &bible_studies_gnd_numbers,
+                        const std::unordered_set<std::string> &canon_law_gnd_numbers) {
+    static std::vector<std::string> tags_to_check{ "100", "110", "111", "700", "710", "711" };
+    unsigned record_count(0);
+    while (const MARC::Record record = marc_reader->read()) {
+        ++record_count;
+        for (auto tag_to_check : tags_to_check) {
+            for (auto &field : record.getTagRange(tag_to_check)) {
+                const auto subfields(field.getSubfields());
+                for (const auto &subfield : subfields) {
+                    if (subfield.code_ == '0') {
+                        const std::string author(subfield.value_);
+                        if (likely(not author.empty())) {
+                            if (not StringUtil::StartsWith(author, "(DE-627)"))
+                                continue;
+                            std::string author_id(author.substr(std::string("(DE-627)").length()));
+                            auto it_author = authors->find(author_id);
+                            std::set<std::string> instances;
+                            if (it_author != authors->end())
+                                instances = it_author->second;
+                            
+                            if (IsRelBibRecord(record))
+                                instances.emplace("r");
+                            if (IsCanonLawRecord(record, canon_law_gnd_numbers))
+                                instances.emplace("c");
+                            if (IsBibleStudiesRecord(record, bible_studies_gnd_numbers))
+                                instances.emplace("b");
+                            
+                            if (authors->find(author_id) == authors->end())
```

Comment:
This will replace the value even if it already existed and even if we added the values using the iterator (which shouldn't make an emplace necessary).

How about we turn this around => Create an empty std::set there if the key doesn't exist, and then add all values later using the iterator?